### PR TITLE
Initial Setup

### PR DESCRIPTION
**Pull request by dswhitely1**
Start of my React Clone Project

### CODE REVIEW

File: instagram-app/src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+import data from './dummy-data';
+// Global Styles
+import { GlobalStyle } from './components/styles/Global';
+// Components
+import SearchBar from './components/SearchBar/SearchBar';
+import PostContainer from './components/PostContainer/PostContainer';
+
+class App extends Component {
+  constructor( props ) {
```
Comment (Fractured2K) [inline instagram-app/src/App.jsx:10]: Quick little tip, when using `create-react-app` you no longer need to wrap your state in a `constructor` and call `super()` 

So you can take this
```
 constructor( props ) {
    super( props );
    this.state = {
      posts: [],
      nextCommentId: 0,
    };
```

and convert it to this
```
state = {
   posts: [],
   nextCommentId: 0,
};
```

Right off the bat you can see this is already much more concise and easier to read. The reason you can do this is cause of tool named [babel](https://babeljs.io/). Babel is a transpiler, which means it takes the current code you've written and rewrites it into the same language typically to a different syntax/version that's well supported on different browsers/environments. So when your state isn't wrapped in a constructor and calling super babel automatically adds that code in there so you don't have to.

for example if you write
```
const foo = () => {
  return 'bar'
}
```

babel with transpiles that into
```
var foo = function foo() {
    return 'bar';
}
```

you can see it has the exact same functionality but just a different syntax that is well supported on all web browsers

File: instagram-app/src/components/SearchBar/SearchBar.scss [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.show-sprite {
+  display: block;
+}
+
+.sprite {
```
Comment (Fractured2K) [inline instagram-app/src/components/SearchBar/SearchBar.scss:5]: I really love the implementation of the sprite, it takes the application to a whole new level.

File: instagram-app/src/components/Layout/StyledComponents.jsx [hunk 3]
```diff
@@ -0,0 +1,205 @@
+import styled from 'styled-components';
```
Comment (Fractured2K) [inline instagram-app/src/components/Layout/StyledComponents.jsx:1]: Awesome to see you using styled components already!

**Review by Fractured2K — COMMENTED** [review]
Overall really solid job so far Don! Great work as always keep it up!

File: instagram-app/src/App.jsx (hunk 1 above)
Comment (dswhitely1) [reply instagram-app/src/App.jsx:10]: Thanks for the feedback.  Abstracting State away from the constructor method does clean up the code better, knowing that babel will compile the code to be correct.  I will institute this on the next commit.

Thanks,

Donald Whitely

Comment (Fractured2K) [reply instagram-app/src/App.jsx:10]: Awesome! 🙌 Can't wait to see it!

**Pull request merged**